internal/roomname.go

```diff
@@ -80,9 +80,21 @@ func (m *RoomMetadata) CopyHeroes() *RoomMetadata {
 	newMetadata.Heroes = make([]Hero, len(m.Heroes))
 	copy(newMetadata.Heroes, m.Heroes)
 
-	// ⚠️ NB: there are other pointer fields (e.g. PredecessorRoomID *string) or
-	// and pointer-backed fields (e.g. LatestEventsByType map[string]EventMetadata)
-	// which are not deepcopied here.
+	// copy LatestEventsByType else we risk concurrent map r/w when the connection
+	// reads this map and updates write to it.
+	newMetadata.LatestEventsByType = make(map[string]EventMetadata)
+	for k, v := range m.LatestEventsByType {
+		newMetadata.LatestEventsByType[k] = v
+	}
+
+	newMetadata.ChildSpaceRooms = make(map[string]struct{})
+	for k, v := range m.ChildSpaceRooms {
+		newMetadata.ChildSpaceRooms[k] = v
+	}
+
+	// ⚠️ NB: there are other pointer fields (e.g. PredecessorRoomID *string)
+	// and pointer-backed fields which are not deepcopied here, because they do not
+	// change.
```

Comment:
namely: room type, room predecessor, upgraded room ID.

I think the latter can change if someone sends a second tombstone, but we likely don't care.